jruby-gradle-base-plugin/src/main/groovy/com/github/jrubygradle/internal/mavengem/Handler.java

```diff
@@ -0,0 +1,57 @@
+package com.github.jrubygradle.internal.mavengem;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.net.URL;
+import java.net.URLConnection;
+import java.net.URLStreamHandler;
+
+public class Handler extends URLStreamHandler {
+
+    public static String KEY = "java.protocol.handler.pkgs";
+    public static String PKG = "com.github.jrubygradle.internal";
+    private final String MAVEN_RELEASES = "/maven/releases";
+    private final int MAVEN_RELEASES_LEN = MAVEN_RELEASES.length();
+
+    public synchronized static boolean isMavenGemProtocolRegistered() {
+        return System.getProperties().contains(KEY);
+    }
+
+    public synchronized static boolean registerMavenGemProtocol() {
+        if (System.getProperties().contains(KEY)) {
+            String current = System.getProperty(KEY);
+            if (!current.contains(PKG)) {
+                System.setProperty(KEY, current + "|" + PKG);
+            }
+        }
+        else {
+            System.setProperty(KEY, PKG);
+        }
+        try {
+            URL url = new URL("mavengem:https://rubygems.org/maven/releases/ping");
```

Comment:
Does this mean that in order to register the URL handler correctly the user's host must be able to contact rubygems.org?